File: 02-backend/demoEcommerceWeb/src/main/java/com/Ecom/demoEcommerceWeb/requestmodels/AddProductRequest.java
package com.Ecom.demoEcommerceWeb.requestmodels;

// Importing Lombok's @Data
import lombok.Data;

/**
 * Data transfer object for adding a product to the e-commerce system.
 */
@Data
public class AddProductRequest {

    /**
     * Title of the product.
     */
    private String title;

    /**
     * Seller of the product.
     */
    private String seller;

    /**
     * Description of the product.
     */
    private String description;

    /**
     * Total quantity of the product.
     */
    private int quantity;

    /**
     * Category of the product.
     */
    private String category;

    /**
     * Email of the merchant who sells the product.
     */
    private String merchantEmail;

    /**
     * URL or path to the product's image.
     */
    private String img;
}

File: 03-frontend/react-library/src/models/AddProductRequest.ts
class AddProductRequest {
    title: string;
    seller: string;
    description: string;
    quantity: number;
    category: string;
    merchantEmail?:string;
    img?: string;
    constructor(title:string,seller:string,description:string, quantity:number,category:string){
        this.title=title;
        this.seller=seller;
        this.description=description;
        this.quantity=quantity;
        this.category=category;
        //this.merchantEmail=merchantEmail;
    }
}
export default AddProductRequest;